domains/client/domain-executor/src/parent_chain.rs

```diff
@@ -0,0 +1,142 @@
+use crate::utils::to_number_primitive;
+use sp_api::{NumberFor, ProvideRuntimeApi};
+use sp_blockchain::HeaderBackend;
+use sp_domains::fraud_proof::FraudProof;
+use sp_domains::{DomainId, ExecutorApi};
+use sp_runtime::generic::BlockId;
+use sp_runtime::traits::Block as BlockT;
+use std::marker::PhantomData;
+use std::sync::Arc;
+use subspace_core_primitives::BlockNumber;
+use system_runtime_primitives::SystemDomainApi;
+
+/// Trait for interacting between the domain and its corresponding parent chain, i.e. retrieving
+/// the necessary info from the parent chain or submit extrinsics to the parent chain.
+///
+/// - The parent chain of System Domain => Primary Chain
+/// - The parent chain of Core Domain => System Domain
+pub(crate) trait ParentChainInterface<Hash>: Sync + Send {
+    fn head_receipt_number(&self, at: Hash) -> Result<BlockNumber, sp_api::ApiError>;
+    fn maximum_receipt_drift(&self, at: Hash) -> Result<BlockNumber, sp_api::ApiError>;
+    fn submit_fraud_proof_unsigned(&self, fraud_proof: FraudProof) -> Result<(), sp_api::ApiError>;
+}
+
+/// The parent chain of the core domain
+pub struct CoreDomainParentChain<SClient, SBlock, PBlock> {
+    // The system domain client
+    system_domain_client: Arc<SClient>,
+    // The id of the core domain
```

Comment:
I believe `///` is preferred, see https://github.com/subspace/subspace/pull/1019#discussion_r1050306673
```suggestion
    /// Core domain id
```